Build Error Java 25

When building xrpl4j-core using Jdk 25, Immutable classes are not generated.

This behaviour is known from Immutables lib team and is documented (see Maven Classic in https://immutables.github.io/getstarted.html)

By applying their fix (adding compiler args in maven-compiler-plugin) it works.

Sadly by doing this, it doesn't work anymore using Java 8. I didn't opened a PR for this reason, this is not a proper fix.

Yep, felt bad opening a ticket and not take the few minutes it took to fix it...

ITs are mostly in error  when running them locally, but it's regardless of the branch or the jdk.
`[ERROR] Tests run: 105, Failures: 0, Errors: 98, Skipped: 4`

For instance, they are mostly in error even running them on main branch with jdk 8.
Should they be working? Maybe I'm doing something wrong (mvn clean install)

--- a/pom.xml
+++ b/pom.xml
@@ -135,9 +135,9 @@
       </dependency>
       <dependency>
         <groupId>org.immutables</groupId>
-        <artifactId>value</artifactId>
+        <artifactId>value-annotations</artifactId>
         <version>2.9.3</version>
-        <scope>provided</scope>
+        <scope>provided</scope> <!-- compile only -->
       </dependency>
       <dependency>
         <groupId>org.slf4j</groupId>
@@ -331,10 +331,18 @@
         <!-- org.apache.maven.plugins:maven-compiler-plugin -->
         <plugin>
           <artifactId>maven-compiler-plugin</artifactId>
-          <version>3.10.1</version>
+          <version>3.15.0</version>
           <configuration>
             <source>${maven.compiler.source}</source>
             <target>${maven.compiler.target}</target>
+
+            <annotationProcessorPaths>
+              <path>
+                  <groupId>org.immutables</groupId>
+                  <artifactId>value</artifactId> <!-- contains the annotation processor -->
+                  <version>2.9.3</version>
+              </path>
+            </annotationProcessorPaths>
           </configuration>
         </plugin>
 
@@ -434,7 +442,7 @@
         <!-- org.jacoco:jacoc-maven-plugin -->
         <plugin>
           <groupId>org.jacoco</groupId>
-          <version>0.8.10</version>
+          <version>0.8.14</version>
           <artifactId>jacoco-maven-plugin</artifactId>
           <executions>
             <execution>

--- a/xrpl4j-client/pom.xml
+++ b/xrpl4j-client/pom.xml
@@ -47,7 +47,7 @@
     </dependency>
     <dependency>
       <groupId>org.immutables</groupId>
-      <artifactId>value</artifactId>
+      <artifactId>value-annotations</artifactId>
     </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>

--- a/xrpl4j-core/pom.xml
+++ b/xrpl4j-core/pom.xml
@@ -39,7 +39,7 @@
     </dependency>
     <dependency>
       <groupId>org.immutables</groupId>
-      <artifactId>value</artifactId>
+      <artifactId>value-annotations</artifactId>
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>

--- a/xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/NfTokenMint.java
+++ b/xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/NfTokenMint.java
@@ -56,7 +56,7 @@ public interface NfTokenMint extends Transaction {
    * multiple tokens. The implementation reserves taxon identifiers greater
    * than or equal to 2147483648 (0x80000000).
    *
-   * @return Taxon of type {@link int} associated with the Token.
+   * @return Taxon of type {@link UnsignedLong} associated with the Token.
    */
   @JsonProperty("NFTokenTaxon")
   UnsignedLong tokenTaxon();

--- a/xrpl4j-integration-tests/pom.xml
+++ b/xrpl4j-integration-tests/pom.xml
@@ -35,7 +35,7 @@
     </dependency>
     <dependency>
       <groupId>org.immutables</groupId>
-      <artifactId>value</artifactId>
+      <artifactId>value-annotations</artifactId>
       <scope>test</scope>
     </dependency>
     <dependency>

